protocol/msgpack/aggregated_iterator.go

```diff
@@ -46,7 +46,7 @@ func NewAggregatedIterator(reader io.Reader, opts AggregatedIteratorOptions) Agg
 	}
 	return &aggregatedIterator{
 		ignoreHigherVersion: opts.IgnoreHigherVersion(),
-		iteratorBase:        newBaseIterator(reader),
+		iteratorBase:        newBaseIterator(reader, opts.PolicyDecompressionOptions()),
```

Comment:
This might be a little tricky since the interface for the options for the aggregated and unaggregated iterators are different. I added a base iterator options interface which the aggregated and unaggregated iterators can return which might be better here. The one difficulty is that since we set return the options when we set a field the return types will be different for the interfaces even if we are setting the same field because they are returning a different interface.